server/server.js

```diff
@@ -0,0 +1,119 @@
+import express, { response } from 'express';     
+import cors from 'cors';
+const app = express();
+import { join, dirname } from 'path';
+import { Low, JSONFile } from 'lowdb';
+import { fileURLToPath } from 'url';
+import { v4 as uuidv4 } from 'uuid';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const file = join(__dirname, 'data/db.json');
+const adapter = new JSONFile(file);
+const db = new Low(adapter);
+
+await db.read();
+
+if (!db.data) db.data = { users: [] };
+
+app.use(express.json());
+app.use(cors());
+
+
+
+//* SIGN UP a NEW USER
+app.post("/signup", async (request, response)=> {
+console.log("new user signing up!");
+
+let signupNew = {id: uuidv4(), ...request.body};
+
+if(!request.body.username) {
+    return response.json(
+       {error: "what do you want? bring some data or go away"} 
+    )
+}
+
+const duplicate = db.data.users.find(user => user.username === request.body.username)
+
+if (duplicate){
+    return response.json(
+       {error: `${request.body.username} is already taken, sorry`} 
+    )
+}
+
+db.data.users.push(signupNew);
+await db.write();
+response.json(signupNew);
+})
+
+
+
+//* USER LOGIN
+app.post("/login", (request, response)=> {
+    console.log("logging in now...")
+
+const {username, password} = request.body;
+
+const certifiedUser = db.data.users.find(user => user.username === username && user.password === password);
+
+    if (!certifiedUser) {
+    console.log("log in failed");
+    response.json("invalid user");
```

Comment:
The json() needs to get passed JavaScript data. 
So ideally we would send back some object with a message here, e.g.
`response.json( { error: "invalid user" } ) ` just like you did in your signup route correctly :+1: